micrometer-core/src/main/java/io/micrometer/core/instrument/binder/undertow/UndertowMetrics.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.undertow;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.TimeGauge;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+import io.undertow.server.handlers.MetricsHandler;
+import io.undertow.servlet.api.MetricsCollector;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToLongFunction;
+
+import static io.undertow.server.handlers.MetricsHandler.MetricResult;
+
+/**
+ * {@link MetricsHandler} to collect metrics for {@link io.undertow.Undertow}
+ *
+ * @author Dharmesh Jogadia
+ * */
+@NonNullApi
+@NonNullFields
+public class UndertowMetrics implements MetricsCollector, MeterBinder {
```

Comment:
Why not implement it as more general `MetricsHandler`? So it can be used outside of servlet api (which more generic)

Reply:
Ok, I'll check and get back, if there's another way to scrap undertom metrics. 
Any more suggestions or references are welcome...!

Reply:
I did some research on Undertow generic metrics and here a mail [thread](https://lists.jboss.org/pipermail/undertow-dev/2017-October/002123.html) I found from undertow-dev.

Reply:
@IRus is the idea that we would use the Undertow `MetricsHandler` somehow, or would we make our own `HttpHandler` implementation that does something similar but with Micrometer?

Reply:
[MetricsHandler](https://github.com/undertow-io/undertow/blob/master/core/src/main/java/io/undertow/server/handlers/MetricsHandler.java) doesn't preserve request paths and methods, so yes, I think will be better to have in-house one

Reply:
@dharmeshjogadia would that work for the metrics you want to make available for Undertow? If so, would you be willing to update this pull request with that implementation?
I notice you're tagging the servlet name in the current implementation. I wonder if we would be able to get that kind of information from an `HttpHandler` implementation. If not, maybe we will need the current implementation in addition to an `HttpHandler` one - but users would have to be careful to not configure both if they're using servlet.

Reply:
Still looking for feedback on my previous comment if folks have time.